wsmaster/codenvy-hosted-api-organization/src/main/java/com/codenvy/organization/spi/impl/OrganizationImpl.java

```diff
@@ -0,0 +1,144 @@
+/*
+ *  [2012] - [2016] Codenvy, S.A.
+ *  All Rights Reserved.
+ *
+ * NOTICE:  All information contained herein is, and remains
+ * the property of Codenvy S.A. and its suppliers,
+ * if any.  The intellectual and technical concepts contained
+ * herein are proprietary to Codenvy S.A.
+ * and its suppliers and may be covered by U.S. and Foreign Patents,
+ * patents in process, and are protected by trade secret or copyright law.
+ * Dissemination of this information or reproduction of this material
+ * is strictly forbidden unless prior written permission is obtained
+ * from Codenvy S.A..
+ */
+package com.codenvy.organization.spi.impl;
+
+import com.codenvy.organization.shared.model.Organization;
+import com.codenvy.organization.spi.jpa.OrganizationEntityListener;
+
+import org.eclipse.che.account.spi.AccountImpl;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.EntityListeners;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.NamedQueries;
+import javax.persistence.NamedQuery;
+import javax.persistence.OneToOne;
+import java.util.Objects;
+
+/**
+ * Data object for {@link Organization}.
+ *
+ * @author Sergii Leschenko
+ */
+@Entity(name = "Organization")
+@EntityListeners(OrganizationEntityListener.class)
+@NamedQueries(
+        {
+                @NamedQuery(name = "Organization.getByName",
+                            query = "SELECT o " +
+                                    "FROM Organization o " +
+                                    "WHERE o.account.name = :name"),
```

Comment:
We have unique index there https://github.com/eclipse/che/blob/jpa-integration/wsmaster/che-core-api-account/src/main/java/org/eclipse/che/account/spi/AccountImpl.java#L41-41